app-frontend/src/containers/MainPage/NavigationContainer.tsx

```diff
@@ -1,27 +1,25 @@
-import React, { useContext } from 'react';
+import React, { FC } from 'react';
+import { useCookies } from 'react-cookie';
 import { useNavigate } from 'react-router-dom';
-import AuthContext from '../../context/AuthContext';
-import useAuth from '../../hooks/auth.hook';
+import useTypedSelector from '../../hooks/typedSelector.hook';
 import Navigation from '../../components/MainPage/Navigation';
-import initialState from '../../store/initialState';
 
-const NavigationContainer = function () {
-  const auth = useContext(AuthContext);
+const NavigationContainer:FC = function () {
   const navigate = useNavigate();
-  const { logout } = useAuth();
-  const isAuthenticated:boolean = !initialState.user.data;
+  const [cookies, setCookie, removeCookie] = useCookies(['user']); // eslint-disable-line
+  const { data } = useTypedSelector((state) => state.user);
 
   const handleLogout = (): void => {
-    logout();
+    removeCookie('user');
```

Comment:
all work with side effects should be moved into sagas.
For you should be only 1 source of truth. When you work with cookies directly from components, you ruin this principle of redux.